module/proto/gravity/v1/gravity.proto

```diff
@@ -0,0 +1,92 @@
+syntax = "proto3";
+package gravity.v1;
+
+import "cosmos_proto/cosmos.proto";
+import "google/protobuf/timestamp.proto";
+import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/cosmos/gravity-bridge/module/x/gravity/types";
+
+// EthereumEventVoteRecord is an event that is pending of confirmation by 2/3 of the signer set.
+// The event is then attested and executed in the state machine once the required
+// threshold is met.
+message EthereumEventVoteRecord {
+  google.protobuf.Any event    = 1 [
+    (cosmos_proto.accepts_interface) = "EthereumEvent"
+  ];  
+  repeated string     votes    = 2;
+  bool                accepted = 3;
+}
+
+// LatestEthereumBlockHeight defines the latest observed ethereum block height and the
+// corresponding timestamp value in nanoseconds.
+message LatestEthereumBlockHeight {
+  uint64 ethereum_height = 1;
+  uint64 cosmos_height   = 2;
+}
+
+// EthereumSigner represents a cosmos validator with its corresponding bridge operator
+// ethereum address and its staking consensus power.
+message EthereumSigner {
+  int64  power            = 1;
+  string ethereum_address = 2;
+}
+
+// UpdateSignerSetTx is the Ethereum Bridge multisig set that relays transactions
+// the two chains. The staking validators keep ethereum keys which are used to
+// check signatures on Ethereum in order to get significant gas savings.
+message UpdateSignerSetTx {
+  uint64             nonce   = 1;
+  repeated EthereumSigner signers = 2 [
+    (gogoproto.castrepeated) = "EthereumSigners", 
+    (gogoproto.nullable) = false
+  ];
+}
+
+// BatchTx represents a batch of transactions going from Cosmos to Ethereum. Batch
+// txs are are identified by a unique hash and the token contract that is shared by
+// all the SendToEthereum
+message BatchTx {
+  uint64         nonce          = 1;
+  uint64         timeout        = 2;
+  repeated bytes transactions   = 3 [
+    (gogoproto.casttype) = "github.com/tendermint/tendermint/libs/bytes.HexBytes"
+  ];
+  string         token_contract = 4;
+  uint64         ethereum_block = 5;
+}
+
+// SendToEthereumTx represents an individual SendToEthereum from Cosmos to Ethereum
+message SendToEthereumTx {
+  uint64                   id                 = 1;
```

Comment:
nit, I think it would be great to add comments on top of each of the proto messages fields. I wrote a bunch of them on the old refactor branch that you can port over 👍

Reply:
Do comments on the individual fields generally go over the field in question, or on top of the whole message?

Reply:
I want to get agreement on the structures before we start adding comments.

Reply:
> Do comments on the individual fields generally go over the field in question, or on top of the whole message?

fields description on top of each of the fields